estate/models/estate_property.py

```diff
@@ -0,0 +1,99 @@
+from odoo import api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.float_utils import float_compare, float_is_zero
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Estate Property"
+    _sql_constraints = [
+        ("nonnegative_property_expected_price", "CHECK(expected_price >= 0)", "The expected price of a property can't be negative."),
+        ("nonnegative_property_selling_price", "CHECK(selling_price >= 0)", "The selling price of a property can't be negative."),
+    ]
+
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    salesperson_id = fields.Many2one("res.users", string="Salesperson", default=lambda self: self.env.user)
+    buyer_id = fields.Many2one("res.partner", string="Buyer", copy=False)
+    tag_ids = fields.Many2many("estate.property.tag", string="Property Tags")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+
+    name = fields.Char(required=True)
```

Comment:
Why separate the relational fields to the others ?

Reply:
Just thought this would be more readable and easier to manage model relations.